resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/builders/RefreshLocalJavaFileBuilder.java

```diff
@@ -36,18 +40,17 @@ protected IProject[] build(int kind, Map<String, String> args, IProgressMonitor
 		IFile file = project.getFile("A.java");
 		IPath localLocation = project.getLocation().append(file.getName());
 		java.io.File localFile = localLocation.toFile();
-		FileOutputStream out = null;
-		try {
+		try (FileOutputStream out = new FileOutputStream(localFile)) {
 			if (localFile.exists()) {
```

Comment:
At this point we have an open output stream on `localFile` and we delete the file before we start writing to it. 
Previously it used to be the case that the stream was made after deleting the file.
If I try this locally, the file doesn't exist after `out` was closed:

```java
		File tempDir = Files.createTempDir();
		File localFile = new File(tempDir, "a.txt");
		localFile.createNewFile();
		try (FileOutputStream out = new FileOutputStream(localFile)) {
			if (localFile.exists()) {
				localFile.delete();
			}
			out.write("public class A {}".getBytes());
		}
		try (FileInputStream in = new FileInputStream(localFile)) {
			in.readAllBytes();
		}
```

Reply:
thanks -> https://github.com/eclipse-platform/eclipse.platform/pull/1129